app/server/src/main/scala/org/bitcoins/server/BitcoinSServerMain.scala

```diff
@@ -54,6 +56,13 @@ class BitcoinSServerMain(override val serverArgParser: ServerArgParser)(implicit
 
     for {
       _ <- startedConfigF
+
+      _ <-
+        if (torConf.enabled) {
+          val tor = torConf.createClient
+          tor.startBinary()
```

Comment:
What if tor daemon is already up and running (by `systemd`, for example)? Do we want to start it anyway?

Probably, first we should check if the ports defined in `torConf` are already bound, and if not, then start the binary.

Reply:
currently it will attempt to start up but fail if it is already running

```
Aug 09 12:29:22.656 [notice] Tor 0.4.2.7 running on Linux with Libevent 2.1.11-stable, OpenSSL 1.1.1f, Zlib 1.2.11, Liblzma 5.2.4, and Libzstd 1.4.4.
Aug 09 12:29:22.656 [notice] Tor can't help you if you use it wrong! Learn how to be safe at https://www.torproject.org/download/download#warning
Aug 09 12:29:22.657 [notice] Read configuration file "/etc/tor/torrc".
Aug 09 12:29:22.660 [warn] CookieAuthFileGroupReadable is set, but will have no effect: you must specify an explicit CookieAuthFile to have it group-readable.
Aug 09 12:29:22.660 [notice] Opening Socks listener on 127.0.0.1:9050
Aug 09 12:29:22.660 [warn] Could not bind to 127.0.0.1:9050: Address already in use. Is Tor already running?
Aug 09 12:29:22.660 [notice] Opening Control listener on 127.0.0.1:9051
Aug 09 12:29:22.660 [warn] Could not bind to 127.0.0.1:9051: Address already in use. Is Tor already running?
Aug 09 12:29:22.660 [warn] Failed to parse/validate config: Failed to bind one of the listener ports.
Aug 09 12:29:22.660 [err] Reading config failed--see warnings above.
```

Reply:
Since we start the binary here, don't we need to stop it as well? Otherwise a users machine when the JVM gets killed will still have tor running